### PR TITLE
2024 02 20 tranche mirror

### PR DESCRIPTION
**Pull request by thedavidmeister**

### CODE REVIEW

File: test/TrancheSpreadTest.t.sol [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+        ffi[29] = "--bind";
+        ffi[30] = string.concat("tranche-space-edge-guard-threshold=", uint2str(TRANCHE_EDGE_THRESHOLD));
+        
+        
+        trancheRefill = bytes.concat(getSubparserPrelude(orderBookSubparser), vm.ffi(ffi));
+    }
+
+    function getSubparserPrelude(address obSubparser) internal pure returns (bytes memory) {
+        bytes memory RAINSTRING_OB_SUBPARSER =
+            bytes(string.concat("using-words-from ", obSubparser.toHexString(), " "));
```
Comment (thedavidmeister) [inline test/TrancheSpreadTest.t.sol:513]: didn't we update dotrain to be able to support this in the .rain file? @Siddharth2207 @rouzwelt

File: test/TrancheSpreadTest.t.sol [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+        return RAINSTRING_OB_SUBPARSER;
+    }
+    
+    function uint2str(uint256 _i) internal pure returns (string memory _uintAsString) {
```
Comment (thedavidmeister) [inline test/TrancheSpreadTest.t.sol:517]: this already exists in open zeppelin, which is a dep of this repo

File: test/TrancheSpreadTest.t.sol (hunk 1 above)
Comment (Siddharth2207) [reply test/TrancheSpreadTest.t.sol:630]: Do we give the subparser address directly in the .rain file ? Or a binding? Binding doesn't seem to work with current dotrain

File: test/TrancheSpreadTest.t.sol (hunk 2 above)
Comment (Siddharth2207) [reply test/TrancheSpreadTest.t.sol]: resolved

**Pull request merged**